### ISSUE TITLE
Allow passing arguments to run-python3-tests

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7234]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#7234|
|Type|enhancement|
|Created|2014-04-22 10:33:21Z|
|Branch|https://github.com/twisted/twisted/tree/run-python3-argv-7234|

run-python3-test is hardcoded to '-v' arguments.

When doing development work sometimes I wanted to interupt the tests with ctrl + c and for this unittest needs '-c' argument.

Attachments:

* [7234-1.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/64a/64a7535ace5b9ed811c40df45c7081260240ce42/a9449e4147e42592c48578a78f9fa2da5a69e913.diff) (1381 bytes) - added by adiroiban on 2014-04-22 10:38:09Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__7234 7234
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__ 
branch__branches_run_python3_argv_7234 branches/run-python3-argv-7234
branch_author__glyph glyph
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1398162801000000 1398162801000000
changetime__1415730616512823 1415730616512823
version__None None
owner__glyph glyph

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban commented|
|-|-|

I have attached a patch which allows passing argument to run-python3-tests.


```
./admin/run-python3-tests
```

Will run all default tests with -v


```
./admin/run-python3-tests test_name
```

Will run test_name with -v

```
./admin/run-python3-tests -c
```


Will still run all default tests but with -c argument and without -v argument

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph commented|
|-|-|

(In [43448]) Branching to run-python3-argv-7234.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph set owner to @glyph|
|-|-|

Since this script is in `admin/`, has no test coverage so far, and, god willing, we will delete it soon, I am going to go ahead and commit this despite the lack of test coverage.  I've manually verified that all the same tests are still being run by the buildbot.  I _want_ to say something like "but you should have spent the time working on trial instead", except I know effort is not fungible, and I suspect that just committing this and getting it out of the way will actually accelerate trial being available on py3 :).

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set status to `closed`|
|-|-|

(In [43451]) Merge run-python3-argv-7234: Allow options to run-python3-tests

Author: adiroiban

Reviewer: glyph

Fixes: [#7234](https://github.com/twisted/twisted/issues/7234)

Previously run-python3-tests was hard-coded to take only `-v`.  Now it
can take other options (such as `-c` to allow interrupting the tests
with `C-c`).
